snuba/consumers/consumer_builder.py

```diff
@@ -62,6 +65,39 @@ class MockParameters:
     std_deviation: int
 
 
+def validate_passed_slice(
+    storage_key: StorageKey, slice_id: Optional[int] = None
+) -> None:
+    # make sure this is a storage that is sliced
+    # and that the slice id passed in is valid
+    if slice_id is not None:
+        assert storage_key.value in SLICED_STORAGES
+        assert slice_id < SLICED_STORAGES[storage_key.value]
```

Comment:
I think all of this validation could also be part of the centralised KafkaTopicSpec implementation rather than requiring each piece of code to do it on their own

Reply:
This seems like it's no longer about Kafka topic validation -- I'm guessing Lyn's comment no longer applies?

Reply:
Yes. Although I would still encourage you to think about whether you think this code should be reusable and part of the common slice validation logic rather than specific to the consumer builder.